### PR TITLE
[indexer] Leave the most important types in FeatureParams::FinishAddingTypes()

### PR DESCRIPTION
**Pull request by tatiana-yan**
https://jira.mail.ru/browse/MAPSME-10276

### CODE REVIEW

File: indexer/feature_data.cpp [hunk 1]
```diff
@@ -500,10 +500,17 @@ bool FeatureParams::FinishAddingTypes()
   sort(newTypes.begin(), newTypes.end());
   newTypes.erase(unique(newTypes.begin(), newTypes.end()), newTypes.end());
```
Comment (mpimenov) [inline indexer/feature_data.cpp]: SortUnique

File: indexer/feature_data.cpp [hunk 2]
```diff
@@ -500,10 +500,17 @@ bool FeatureParams::FinishAddingTypes()
...
 
-  m_types.swap(newTypes);
+  if (newTypes.size() > kMaxTypesCount)
+  {
+    // Put common types to the end to leave the most important types.
+    static UselessTypesChecker checker;
+    UNUSED_VALUE(base::RemoveIfKeepValid(newTypes.begin(), newTypes.end(),
```
Comment (mpimenov) [inline indexer/feature_data.cpp]: nit: Can we just pass checker as 3rd param?

File: indexer/feature_data.cpp (hunk 1 above)
Comment (tatiana-yan) [reply indexer/feature_data.cpp]: done

File: indexer/feature_data.cpp (hunk 2 above)
Comment (tatiana-yan) [reply indexer/feature_data.cpp]: done

**Comment by tatiana-yan** [comment]
JTALL

**Pull request merged**